serverless.yml

```diff
@@ -0,0 +1,112 @@
+
+
+service: users-api
+
+provider:
+  name: aws
+  runtime: nodejs10.x
+  iamRoleStatements:
+    - Effect: Allow
+      Action:
+        - dynamodb:Query
+        - dynamodb:GetItem
+        - dynamodb:PutItem
+        - dynamodb:UpdateItem
+        - dynamodb:DeleteItem
+      Resource:
+        - arn:aws:dynamodb:${opt:region, self:provider.region}:*:table/users
+
+
+package:
+  exclude:
+    - .dynamodb/**
+    - coverage/**
+    - src/**
+    - test/**
+    - '*.md'
+    - '*.js'
+    - '*.json'
+    - '*.lock'
+  include:
+    - node_modules
+    - dist/**/*.js
+
+custom:
+  dynamodb:
+    stages:
+      - local
+    start:
+      port: 8000
+      inMemory: true
+      heapInitial: 200m
+      heapMax: 1g
+      migrate: true
+      seed: true
+      convertEmptyValues: true  
+
+functions:
+  getUser:
+    handler: dist/index.getUser
+    events:
+      - http:
+          path: users/{id}
+          method: get
+          request:
+            parameters:
+              paths:
+                id: true
+
+  createUser:
+    handler: dist/index.createUser
+    events:
+      - http:
+          path: users
+          method: post
+          request:
+            schema:
+              application/json: ${file(dist/schema/userCreate.json)} 
```

Comment:
> How would you handle custom error messages for the schemas that are handled by API GW? 

I'll be completely honest here and admit that I'm not 100% sure.  This is an area where I think there has been an explosion of growth lately.  Even with API gateway I think there are several ways to validate so I'm not sure if perhaps your OAS spec could instruct API gateway on it's validation and even messaging with replacement params.  That would take some more proofing.

> Any benefit to moving this validation to within the Lambda code itself?  

Certainly if you move the validation inside lambda then you have more flexibility.  If you run into a wall with API Gateway where you are unable to do something (like you said custom messages, conditional validation, pure business logic) then  you can certainly accomplish it in the code.   This also ensures that ALL error handling is in one place and can be made to be consistent.  

This may be a bit outside the bounds of this question but I had, at one point at NS8, implemented a generator.  It would take my Swagger 2 (OAS 3 was not available yet) and on commit of my swagger repo would generate models along with some manipulation of the swagger spec to make the JsonShema compliant to use with my model validation.  The idea was that it would eventually generate clients as well (node, php, java, python).  I do believe this is probably even easier now with OAS 3 and supported tooling and would likely be worth looking into.